### PR TITLE
fix(component): preventing registering metrics twice

### PR DESCRIPTION
**Pull request by jlongo-encora**
### Description of change
- Updated node-key-metrics to check if metrics are already registered in the prom-client global registry before registry them. An exception `A metric with the name x has already been registered.` can happen for example when using Next.js modules loads asynchronously, and more than one loaded in a short space of time
- Exposed `getContentType` from `prom-client`
- Fixed lint issues
- Added tests

### CODE REVIEW

File: src/__tests__/metrics.test.ts [hunk 1]
```diff
@@ -8,21 +8,52 @@ const params = {
...
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenLastCalledWith(params.name)
+    expect(result1).toEqual(result2)
+  })
 
-  it('collects metrics', () => {
-    collect()
+  it('collects metrics', async () => {
+    await collect()
+    // eslint-disable-next-line @typescript-eslint/unbound-method
```
Comment (ignatospadov) [inline src/__tests__/metrics.test.ts:47]: why is this needed?

File: src/__tests__/metrics.test.ts [hunk 2]
```diff
@@ -8,21 +8,52 @@ const params = {
...
+    // @ts-expect-error (ts2339)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    register.getSingleMetric.mockReturnValue(result1)
+    const result2 = method(params)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenCalledTimes(2)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenLastCalledWith(params.name)
+    expect(result1).toEqual(result2)
+  })
```
Comment (ignatospadov) [inline src/__tests__/metrics.test.ts:43]: Does it make sense to split these too?

**Review by ignatospadov — CHANGES_REQUESTED** [review]

File: src/__tests__/metrics.test.ts (hunk 2 above)
Comment (jlongo-encora) [reply src/__tests__/metrics.test.ts:43]: I don't think so, because the test says "prevents to register duplicate metrics", so we need to call it twice

File: src/__tests__/metrics.test.ts (hunk 1 above)
Comment (jlongo-encora) [reply src/__tests__/metrics.test.ts:47]: pre-commit lint was failing. We need to add this due to the mock created

Comment (ignatospadov) [reply src/__tests__/metrics.test.ts:47]: Why was it failing?

File: src/__tests__/metrics.test.ts [hunk 3]
```diff
@@ -8,21 +8,52 @@ const params = {
...
+    [summary, Summary]
+  ])('%p passes correct params to %p', (a, b) => {
+    // @ts-expect-error (ts2339)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    register.getSingleMetric.mockReturnValue(undefined)
+    a(params)
+    expect(b).toHaveBeenCalledWith(params)
+  })
+
+  test.each([[counter], [histogram], [gauge], [summary]])('prevents to register duplicate metrics, method=%p', (method) => {
```
Comment (ignatospadov) [inline src/__tests__/metrics.test.ts:24]: `prevents to register duplicate metrics`
to
`prevents registration of duplicate metrics`

File: src/__tests__/metrics.test.ts (hunk 2 above)
Comment (ignatospadov) [reply src/__tests__/metrics.test.ts:43]: Ok I get it now. I think the test logic can be simplified. What if we setup 2 calls and then assert that the `register.getSingleMetric` was only called once?

Im not trying to be picky but I feel like this test can be simplified. Wdyt? @jlongo-encora

**Review by ignatospadov — CHANGES_REQUESTED** [review]

**Comment by ignatospadov** [comment]
Have we tested to ensure this wont break existing versions? Like in [slug resolution service](https://github.com/cheddartv/slug-resolution-service-news12/blob/main/src/app.ts#L9) @jlongo-encora

File: src/__tests__/metrics.test.ts (hunk 1 above)
Comment (jlongo-encora) [reply src/__tests__/metrics.test.ts:47]: This is due to new attributes in the mocks I believe

File: src/__tests__/metrics.test.ts (hunk 2 above)
Comment (jlongo-encora) [reply src/__tests__/metrics.test.ts:43]: @ignatospadov
When calling twice `counter(params)` with the same `params.name`, `register.getSingleMetric` will be called twice. We do need to ensure the return objects are the same.

File: src/__tests__/metrics.test.ts (hunk 3 above)
Comment (jlongo-encora) [reply src/__tests__/metrics.test.ts:24]: Will adjust this
